sdk/monitor/monitor-query/test/internal/unit/metricsQueryClient.unittest.spec.ts

```diff
@@ -0,0 +1,33 @@
+import { DefaultAzureCredential } from "@azure/identity";
+import { assert } from "@azure/test-utils";
+import { Durations, MetricsQueryClient } from "../../../src";
+
+it("verify tracing", async () => {
+    const credential = new DefaultAzureCredential();
+    const client = new MetricsQueryClient(credential, {
+      endpoint: "https://customEndpoint1",
+    });
+    await assert.supportsTracing(
+      async (options) => {
+        const promises: Promise<any>[] = [
+         client.queryResource("resourceId",["metricName1","metricName2"],{
+            granularity: "PT1M",
+            timespan: { duration: Durations.fiveMinutes },
+            ...options
+         }),
+         //Type 'PagedAsyncIterableIterator<MetricDefinition, MetricDefinition[], PageSettings>' is missing the following properties from type 'Promise<any>': then, catch, finally, [Symbol.toStringTag]
+        //  await client.listMetricDefinitions("resourceUri").byPage({
+        //      maxPageSize: 2,
+        //      continuationToken: null
+        //  }),
+        // client.listMetricNames("resourceUri")
```

Comment:
Unsure about the accurate way of testing the list methods with the `supportsTracing` . I put the error in the comment above  cc: @maorleger @xirzec

Reply:
I think if you call `next` it should work. `byPage` itself is not an async function and doesn't make any calls. `next` starts fetching data